### PR TITLE
Gems and rings

### PR DESCRIPTION
**Pull request by AEFeinstein**
### Description

Pretty much the whole game

### Test Instructions

* Play a game against someone else
* Unlock new markers
* Select different markers
* Read the instructions
* Check your record

### Ticket Links

#233

### Readiness Checklist
- [x] I have run `make format` to format the changes
- [x] I have compiled the firmware and the changes have no warnings
- [x] I have compiled the emulator and the changes have no warnings
- [x] I have run `make cppcheck` and checked that `cppcheck_result.txt` has no warnings for the changes
- [x] I have added doxygen comments to any code used by more than one Swadge mode. This includes `/*! \file` comments with Design Philosophy, Usage, and Example sections for new headers.
- [x] I have run `make docs` and checked that `doxy_warnings.txt` has no warnings for the new code

### CODE REVIEW

**Comment by dylwhich** [comment]
I'm out of town without my laptop right now but I should be able to take a proper look at this on Thursday

**Comment by AEFeinstein** [comment]
I had a neighbor demo it and almost all the feedback went to the instructions, so that's getting overhauled

File: main/modes/games/ultimateTTT/ultimateTTTgame.c [hunk 1]
```diff
@@ -14,143 +26,182 @@ typedef void (*cursorFunc_t)(ultimateTTT_t* ttt);
...
+    ttt->game.selectedSubgame.x = 0;
+    ttt->game.selectedSubgame.y = 0;
+    ttt->game.cursorMode        = SELECT_SUBGAME;
+
+    // Default indices
+    ttt->game.p1MarkerIdx = 0;
+    ttt->game.p2MarkerIdx = 0;
+
+    // Clean up after showing instructions
+    ttt->game.p2p.cnc.playOrder = NOT_SET;
```
Comment (dylwhich) [inline main/modes/games/ultimateTTT/ultimateTTTgame.c:77]: I think this (or a related change) is breaking multiplayer. When I try to start the game, both sides connect immediately and then both sides show a blue square around the top-left game, which neither swadge is able to move. I did a git bisect and it looks like it started after 59739eb580b6b6a06857343e103f4305b308b7b6. I looked around a bit and don't see anywhere else outside of `ultimateTTThowTo.c` that actually sets it.

File: emulator/src/components/hdw-tft/hdw-tft.c [hunk 2]
```diff
@@ -13,6 +13,8 @@
 // Const variables
 //==============================================================================
 
+#if defined(CNFGOGL)
+// ARGB
```
Comment (dylwhich) [inline emulator/src/components/hdw-tft/hdw-tft.c:17]: I think this is meant to say `RGBA`

File: main/modes/games/ultimateTTT/ultimateTTTgame.c (hunk 1 above)
Comment (dylwhich) [reply main/modes/games/ultimateTTT/ultimateTTTgame.c:77]: And by "it" I mean `playOrder`

**Comment by dylwhich** [comment]
Oh I love the new tutorial!

**Comment by AEFeinstein** [comment]
Multiplayer died for that tutorial

File: main/modes/games/ultimateTTT/ultimateTTTgame.c (hunk 1 above)
Comment (AEFeinstein) [reply main/modes/games/ultimateTTT/ultimateTTTgame.c:77]: It sure is.

File: emulator/src/components/hdw-tft/hdw-tft.c (hunk 2 above)
Comment (AEFeinstein) [reply emulator/src/components/hdw-tft/hdw-tft.c:17]: It sure should

**Review by dylwhich — APPROVED** [review]

**Pull request merged**